libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -469,6 +469,14 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 						_variable.location(),
 						"Transient cannot be used as data location for constant or immutable variables."
 					);
+
+				if (_variable.value())
+					m_errorReporter.declarationError(
+						9825_error,
+						_variable.location(),
+						"Transient storage state variables cannot be initialized."
```

Comment:
```suggestion
						"Transient storage state variables cannot be initialized during contract creation."
```
Even though initialization *should* be clear enough to the end-user, I feel like being even more explicit would make this sounds less ominous. In any case, it's not a big deal and I'm fine with it staying as is.